app/src/main/java/com/devshish/internship/presentation/ui/player/PlayerViewModel.kt

```diff
@@ -3,42 +3,22 @@ package com.devshish.internship.presentation.ui.player
 import androidx.lifecycle.LiveData
 import androidx.lifecycle.MutableLiveData
 import androidx.lifecycle.ViewModel
-import androidx.lifecycle.viewModelScope
 import com.devshish.internship.domain.model.Song
 import com.devshish.internship.presentation.ui.service.client.MediaBrowserClient
-import kotlinx.coroutines.launch
 
 class PlayerViewModel(
     private val mediaBrowser: MediaBrowserClient
 ) : ViewModel() {
 
-    fun toggle() = mediaBrowser.toggle()
-
-    private val song = Song(null, "Pure Souls", "Kanye West", 300, null)
-
     val songToPlay: LiveData<Song>
         get() = _songToPlay
     private val _songToPlay = MutableLiveData<Song>()
 
-    val pauseEvent: LiveData<Unit>
-        get() = _pauseEvent
-    private val _pauseEvent = MutableLiveData<Unit>()
-
-    val playEvent: LiveData<Unit>
-        get() = _playEvent
-    private val _playEvent = MutableLiveData<Unit>()
-
     init {
-        viewModelScope.launch {
-            _songToPlay.value = song
-        }
-    }
-
-    fun onPause() {
-        _pauseEvent.value = Unit
+        _songToPlay.value = mediaBrowser.getSong()
```

Comment:
It is better to use here callback instead of just asking for song from mediaBrowser